packages/coordinator/src/services/verification/signature-verification.ts

```diff
@@ -114,6 +114,20 @@ function extractSignatureDateTimeStamp(parentPrescriptions: hl7V3.ParentPrescrip
   return new Date(convertHL7V3DateTimeToIsoDateTimeString(author.time))
 }
 
+function extractSignatureMethodFromSignatureRoot(signatureRoot: ElementCompact) {
+  const re = /(rsa-sha)\w+/g
+  const signatureMethod = signatureRoot.Signature.SignedInfo.SignatureMethod._attributes.Algorithm.match(re)
+  if (signatureMethod) {
+    return signatureMethod[0] === "rsa-sha256" ? signatureMethod[0] : "rsa-sha1"
+  } else {
+    signatureRoot.Signature.SignedInfo.SignatureMethod._attributes.Algorithm =
+    "http://www.w3.org/2000/09/xmldsig#rsa-sha1"
+    signatureRoot.Signature.SignedInfo.Reference.DigestMethod._attributes.Algorithm =
+    "http://www.w3.org/2000/09/xmldsig#sha1"
```

Comment:
Should this have xmlenc instead of xmldsig?

Reply:
As discussed the xmldsig value is correct here.